src/hooks/useDomainAvailability.ts

```diff
@@ -0,0 +1,64 @@
+import { useQuery } from 'react-query';
+import { useBuyDomain } from './useBuyDomain';
+import { useZnsSdk } from './useZnsSdk';
+import { MIN_DOMAIN_NAME_LENGTH } from '../constants/domains';
+
+export interface UseDomainAvailabilityReturn {
+	isLoading: boolean;
+	isDomainAvailable: boolean;
+	isDomainSearchEnabled: boolean;
+	domainPrice?: string;
+}
+
+export const useDomainAvailability = (): UseDomainAvailabilityReturn => {
```

Comment:
Can we change the name of this hook, since it also returns the price of the domain?